structure/signprot_modeling.py

```diff
@@ -438,7 +475,7 @@ def run(self):
             self.trimmed_residues.append('G.HG.14')
             self.trimmed_residues.append('G.HG.16')
             self.trimmed_residues.append('G.HG.17')
-        if structure_signprot!=self.target_signprot or alt_signprot_complex.protein not in [None, self.target_signprot]:
+        if structure_signprot!=self.target_signprot or ("alt_signprot_complex" in locals() and alt_signprot_complex.protein not in [None, self.target_signprot]):
```

Comment:
A cleaner alternative could be to initialize "alt_signprot_complex" as None and check if it has changed.